pkg/utils/tar.go

```diff
@@ -138,34 +150,69 @@ func Untar(srcFile string, destination string) error {
 		return err
 	}
 	defer r.Close()
-	tr := tar.NewReader(r)
+	return ExtractFromTar(r, destination)
+}
+
+// ExtractFromTar extracts files from a tar reader into the destination directory
+func ExtractFromTar(u io.Reader, destination string) error {
+	// path inside tar is relative
+	pathBuilder := func(oldPath string) string {
+		return path.Join(destination, oldPath)
+	}
+	tarReader := tar.NewReader(u)
 	for {
-		header, err := tr.Next()
-		switch {
-		case err == io.EOF:
-			return nil
-		case err != nil:
-			return err
-		case header == nil:
-			continue
+		header, err := tarReader.Next()
+		if errors.Is(err, io.EOF) {
+			break
+		}
+		if err != nil {
+			return fmt.Errorf("ExtractFromTar: Next() failed: %w", err)
 		}
-		target := filepath.Join(destination, header.Name)
 		switch header.Typeflag {
 		case tar.TypeDir:
-			if _, err := os.Stat(target); err != nil {
-				if err := os.MkdirAll(target, 0755); err != nil {
-					return err
-				}
+			if err := os.MkdirAll(pathBuilder(header.Name), 0755); err != nil {
```

Comment:
why 0755?

Reply:
You are right, it's better to take the permission from the original file.